server/src/main/scala/org/http4s/server/middleware/AutoSlash.scala

```diff
@@ -29,4 +29,10 @@ object AutoSlash {
       }
     }
   }
+
+  def httpRoutes[F[_]: Monad](httpRoutes: HttpRoutes[F]): HttpRoutes[F] =
+    apply(httpRoutes)
+
+  def httpApp[F[_]: MonoidK: Functor](httpApp: HttpApp[F]): HttpApp[F] =
```

Comment:
I am not sure how valuable this constructor is, given the fact that `F` still should have a `MonoidK` instance and in this case `F` is usually `IO` which  - at least as far as I know - doesn't have that.

Reply:
Agreed. this constructor is not useful

Reply:
Does it need to be a `MonoidK`, or just a `SemigroupK`?  `IO` has one of those (as seen below), though whether it's useful is still dubious.  "Try again with the slash if I raise an error," that would say.

Reply:
Oh, I see the `F.empty` in `apply` now.  Yeah, I think we remove this, keep `httpRoutes`, and declare victory.

Reply:
Ok, I'll update this PR asap.